### PR TITLE
[platform/sfp] Add test cases for SFP error status

### PR DESCRIPTION
**Pull request by stephenxs**
### Description of PR

Summary:
Add test cases for SFP error status

Signed-off-by: Stephen Sun <stephens@nvidia.com>

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Add test cases for SFP error status
- Add test cases for sfputil show error-status
- Add platform test for SFP.get_error_description()

#### How did you verify/test it?
Run regression test.

### CODE REVIEW

**Comment by stephenxs** [comment]
@jleveque can you review this PR when you have time? thanks.

File: tests/platform_tests/api/test_sfp.py [hunk 1]
```diff
@@ -181,6 +181,14 @@ def test_get_presence(self, duthosts, enum_rand_one_per_hwsku_hostname, localhos
                     self.expect(presence is True, "Transceiver {} is not present".format(i))
         self.assert_expectations()
 
+    def test_get_error_description(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
+        for i in self.candidate_sfp:
+            error_description = sfp.get_error_description(platform_api_conn, i)
+            if self.expect(error_description is not None, "Unable to retrieve transceiver {} error description".format(i)):
+                if self.expect(isinstance(error_description, str) or isinstance(error_description, unicode), "Transceiver {} error description appears incorrect".format(i)):
+                    self.expect(error_description == "OK", "Transceiver {} is not present".format(i))
+        self.assert_expectations()
```
Comment (jleveque) [inline tests/platform_tests/api/test_sfp.py:190]: Why add this under the `Functions to test methods inherited from DeviceBase class` section? `get_error_description` is defined in SfpBase. Please move under `Functions to test methods defined in SfpBase class` section.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/platform_tests/api/test_sfp.py (hunk 1 above)
Comment (stephenxs) [reply tests/platform_tests/api/test_sfp.py]: Fixed.

**Review by jleveque — APPROVED** [review]

File: tests/platform_tests/sfp/test_sfputil.py [hunk 2]
```diff
@@ -43,6 +43,22 @@ def test_check_sfputil_presence(duthosts, enum_rand_one_per_hwsku_frontend_hostn
...
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    portmap, dev_conn = get_dev_conn(duthost, conn_graph_facts, enum_frontend_asic_index)
+
+    logging.info("Check output of '{}'".format(cmd_sfp_error_status))
+    sfp_error_status = duthost.command(cmd_sfp_error_status)
+    parsed_presence = parse_output(sfp_error_status["stdout_lines"][2:])
+    for intf in dev_conn:
+        assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
+        assert parsed_presence[intf] == "OK", "Interface error status is not 'OK'"
```
Comment (liat-grozovik) [inline tests/platform_tests/sfp/test_sfputil.py:59]: @stephenxs is it correct that you assume all should be OK and if it is not the test should fail? Will it also fail if the API is not supported by the vendor? should we in this case will not fail the test but rather skip it?

File: tests/platform_tests/api/test_sfp.py [hunk 3]
```diff
@@ -477,6 +477,14 @@ def test_power_override(self, duthosts, enum_rand_one_per_hwsku_hostname, localh
                 self.expect(power_override is False, "Transceiver {} power override data is incorrect".format(i))
         self.assert_expectations()
 
+    def test_get_error_description(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
+        for i in self.candidate_sfp:
```
Comment (liat-grozovik) [inline tests/platform_tests/api/test_sfp.py:481]: can you please add a description of what this tests stands for and what is the expected behaviour?

File: tests/platform_tests/sfp/test_sfputil.py [hunk 4]
```diff
@@ -43,6 +43,22 @@ def test_check_sfputil_presence(duthosts, enum_rand_one_per_hwsku_frontend_hostn
         assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
         assert parsed_presence[intf] == "Present", "Interface presence is not 'Present'"
 
+@pytest.mark.parametrize("cmd_sfp_error_status", ["sudo sfputil show error-status", "sudo sfputil show error-status --fetch-from-hardware"])
+def test_check_sfputil_error_status(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index, conn_graph_facts, cmd_sfp_error_status):
+    """
+    @summary: Check SFP error status using 'sfputil show error-status'
```
Comment (liat-grozovik) [inline tests/platform_tests/sfp/test_sfputil.py:49]: it is not just the error-status command but also --fetch-from-hardware, right? please update the description and if possible add more info on input parameters.

**Review by liat-grozovik — APPROVED** [review]

**Review by liat-grozovik — CHANGES_REQUESTED** [review]
as commented

File: tests/platform_tests/sfp/test_sfputil.py (hunk 2 above)
Comment (stephenxs) [reply tests/platform_tests/sfp/test_sfputil.py:59]: Skip if "Not implemented" found in the output.

File: tests/platform_tests/api/test_sfp.py (hunk 3 above)
Comment (stephenxs) [reply tests/platform_tests/api/test_sfp.py:481]: Fixed

File: tests/platform_tests/sfp/test_sfputil.py (hunk 4 above)
Comment (stephenxs) [reply tests/platform_tests/sfp/test_sfputil.py:49]: Fixed

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**